src/App.js

```diff
@@ -39,24 +45,52 @@ class App extends Component {
   }
 
   goToMain = () => {
-    this.setState({view: 'mainPage', currentMovieId: 0})
+    this.setState({ view: 'mainPage', currentMovieId: 0 })
   }
 
   displayMovieDetails = (id) => {
     console.log(id)
-    this.setState({view: 'detailedView', currentMovieId: id})
+    this.setState({ view: 'detailedView', currentMovieId: id })
   }
 
   render() {
     return (
       <main className='main-page'>
+        <Link to='/'>
         <nav className='nav'>
-          <button className='main-logo' onClick={this.goToMain}><img src={logo} className='movie-reel-logo'/>Cinematic</button>
+          <button className='main-logo' onClick={this.goToMain}><img src={logo} className='movie-reel-logo' />Cinematic</button>
         </nav>
+        </Link>
         {this.checkIfLoading()}
         {this.handleIfFailed()}
-        {this.state.view === 'mainPage' && <Movies movieData={this.state.movieData} displayMovieDetails={this.displayMovieDetails}/>}
-        {this.state.view === 'detailedView' && <MovieDetails id={this.state.currentMovieId}/>}
+        <Switch>
+          <Route
+            exact path='/'
+            render={() => {
+              return (
+                <Movies movieData={this.state.movieData} displayMovieDetails={this.displayMovieDetails} />
+              )
+            }}
+          />
+          <Route
+            exact path='/:id'
+            render={({ match }) => {
+              const movie = this.state.movieData.movies.find(movie => {
+                console.log(match.params.id)
+                return movie.id === parseInt(match.params.id.substring(1));
+              })
+              if(!movie) {
+                return (<p className='error'>This Movie Doesn't Exist!</p>)
```

Comment:
We should double check that this is the right way to error handle users changing the URL